streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatter.java

```diff
@@ -266,6 +276,11 @@ private static class PartiallyDeserializedSegmentValue implements SegmentValue {
         private int deserIndex = -1; // index up through which this segment has been deserialized (inclusive)
         private List<TimestampAndValueSize> unpackedReversedTimestampAndValueSizes;
         private List<Integer> cumulativeValueSizes; // ordered same as timestamp and value sizes (reverse time-sorted)
+        private int valuesStartingIndex = -1; // the index of first value in the segment
+        private int currentCumValueSize = 0; // this is for deserializing a segment records in a lazy manner
+        private int currentDeserIndex = 0; // this is for deserializing a segment records in a lazy manner
```

Comment:
`PartiallyDeserializedSegmentValue` already has `cumulativeValueSizes ` and `deserIndex` because it is already lazy, but only support deserialization in one direction (I believe descending time, ie, ascending order inside the `byte[] segmentValue`).

Why do we need to duplicate both variables? Can't we reuse the existing ones (or course, we might need to add a `Order` member to track in which direction we intent to deserialized?)

Don't want to make it unnecessary complex, so if this idea make the code more difficult, happy to keep the new vars, but it seems we might want to change their name to indicate when (ascending vs descending) they are used?

Reply:
> Why do we need to duplicate both variables?

This is what I did in the first place. But it is complicated since other methods, such as `doInsert()` and other methods, are manipulating (`deserIndex` and `cumulativeValueSizes`) them as well. That's why I decided to have my own variables.

Reply:
> it seems we might want to change their name to indicate when (ascending vs descending) they are used?

I use the same variable (my own defined one) for both directions. Since deserialization is either backwards or forward and not both at the same time.